xray/services/summary.go

```diff
@@ -53,6 +55,59 @@ func (ss *SummaryService) GetBuildSummary(params XrayBuildParams) (*SummaryRespo
 	return &summaryResponse, nil
 }
 
+func (ss *SummaryService) GetArtifactSummary(params ArtifactSummaryParams) (*ArtifactSummaryResponse, error) {
+	httpDetails := ss.XrayDetails.CreateHttpClientDetails()
+	servicesutils.SetContentType("application/json", &httpDetails.Headers)
+
+	requestBody, err := json.Marshal(params)
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+
+	url := fmt.Sprintf("%sartifact", ss.getSummeryUrl())
+	resp, body, err := ss.client.SendPost(url, requestBody, &httpDetails)
+	if err != nil {
+		return nil, err
+	}
+	if err = errorutils.CheckResponseStatus(resp, http.StatusOK); err != nil {
+		return nil, errorutils.CheckError(errorutils.GenerateResponseError(resp.Status, utils.IndentJson(body)))
+	}
+	var response ArtifactSummaryResponse
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+	if response.Errors != nil && len(response.Errors) > 0 {
+		return nil, errorutils.CheckErrorf("Getting artifact-summery for artifact: %s failed with error: %s", response.Errors[0].Identifier, response.Errors[0].Error)
+	}
+	return &response, nil
+}
+
+type ArtifactSummaryParams struct {
+	Checksums []string `json:"checksums,omitempty"`
+	Paths     []string `json:"paths,omitempty"`
+}
+
+type ArtifactSummaryResponse struct {
+	Artifacts []Artifact `json:"artifacts,omitempty"`
+	Errors    []Error    `json:"errors,omitempty"`
+}
+
+type Artifact struct {
+	General General `json:"general,omitempty"`
+	Issues  []Issue `json:"issues,omitempty"`
+	// TODO: Create License struct with correct fields for api/v2/summary/artifact endpoint
+	// Licenses []License `json:"licenses,omitempty"`
+}
+
+type General struct {
+	ComponentId string `json:"component_id,omitempty"`
+	Name        string `json:"name,omitempty"`
+	Path        string `json:"path,omitempty"`
+	PkgType     string `json:"pkg_type,omitempty"`
+	Sha256      string `json:"sha256,omitempty"`
+}
```

Comment:
When comparing to [TypeScript](https://github.com/jfrog/jfrog-client-js/blob/master/model/Xray/Summary/General.ts) and [Java](https://github.com/jfrog/xray-client-java/blob/master/src/main/java/com/jfrog/xray/client/services/summary/General.java), you are missing the following properties:

* `Sha1`
* `ParentSha256`

I notice that they are not part of [the documentation](https://www.jfrog.com/confluence/display/XRAY2X/Xray+REST+API#XrayRESTAPI-ArtifactSummary), but I guess the Go lib should document the same API as TypeScript and Java?!